spec/ysh-builtins.test.sh

```diff
@@ -629,3 +629,13 @@ echo $[l2[1]] # NOTE: depends on dict being ordered
 3
 foo
 ## END
+
+#### List->extend()
+var l = List(1:3)
+echo $[len(l)]
+_ l->extend(List(3:6))
+echo $[len(l)]
```

Comment:
maybe add a test for null return here, and for append(), e.g.

```
var x = l->extend(...)
if (x === null) { echo pass }
```

reminds me that we need the `assert` builtin

Reply:
Added `assert()` while I was in here. What do you think?